### ISSUE TITLE
Version 1.17 (20191130002) incredibly slow download rate

### ISSUE THREAD

**Issue by mp1963** [Ascoware/get-iplayer-automator#270]
*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*
Open the app, add a BBC programme to download
*What is the expected output? What do you see instead?*
BBC programme should download quickly, as per the previous versions of GIA
*What version of GiA are you using? On what version of OS X?*
Version 1.17 (20191130002) on MacOS 10.15.2
*Are you using a web proxy? DNS proxy? VPN? VPS?*
No
*What program are you attempting to download (full title, series and episode)?
Rick Stein's Secret France: Series 1 - Episode 6

*Please provide any additional information below.*
It doesn't matter which files you download load, the rate is slow. I have a 400Mb connection and previous versions of GIA performed quickly as you would expect.

**Comment by mutantcamel** [comment]
"Slow" is a pretty subjective.  I'm downloading the episode you mentioned at 22Mb/sec using a VPN in China.

**Comment by mp1963** [comment]
Ok. One 45 minute episode of a documentary from BBC4 took 20 minutes this morning (2.38Gb) .. that used to take seconds

**Comment by skovatch** [comment]
Yes, 22-26Mb/sec is what I'm getting in CA with a smart DNS service and my cable modem.

**Comment by mutantcamel** [comment]
> Ok. One 45 minute episode of a documentary from BBC4 took 20 minutes this morning (2.38Gb) .. that used to take seconds
> […](#)

So that's been the norm for me all year, more than likely something has changed with your ISP or modem, or even the BBC site, and it coincided with the update.  There's nothing GiA or get_iplayer can do about it.

**Comment by whitedb** [comment]
Interestingly, I am experiencing the same "very slow" downloads from the BBC since the 1.17 update under Sierra.  I can confirm that my ISP connection has not changed and other services are not affected.  Today a 2.3GB file took 7mins vs about 2 mins previously.

**Comment by mp1963** [comment]
Are you with Virgin Media by any chance ?

**Comment by whitedb** [comment]
I wish! No, I am on BT.  My download speed is 50Mb/s.

**Comment by JeremyCavanagh** [comment]
I also suffered this problem but, I thought it was a BBC problem until I reverted to 1.16, then I realised what the actual problem is. 

(Just for info, I am one of the lucky few who have a 2Mb/s internet connection with little expectation of high speed access even though work has begun 2 km from where I live.)

The cause is with making applications compatible with macOS 10.15, they then become incompatible with earlier versions. 

I had the same situation with another app that was proudly announced as an improvement and now compatible with 10.15 unfortunately it was completely screwed up for macOS 10.13 which I am running, so downgraded to previous version.

I think developers need to be aware that the behaviours in 10.15 are significantly (and deliberately) different from previous macOS releases. There are two possible solutions that I can see:

1) Have two different versions of the app:
          pre macOS 10.15, probably your biggest user base 
          post macOS 10.15, I expect a much smaller user base but a future growth area.

2) Write the app to be compatible with both, I suspect not easy.

Unfortunately, I suspect that many developers would find both of these a problem as more work is required and that's why many drop support for earlier OS releases regardless of the size of the user base. Homebrew is a classic example, support for macOS Sierra was dropped so I had to upgrade to High Sierra (even though my Mac is not supposed to be compatible. Thanks Apple).

**Comment by sheepworrier** [comment]
I've had something like this for weeks also - seems to be mainly ITV programmes, which get to about 50% at normal speed and then hang.  If I leave them overnight, then sometimes they are completed by morning

**Comment by mutantcamel** [comment]
I just want to clarify things in my head: I'm running 10.13.6, which is High Sierra and the last official update that my Late-2011 MacBook will see after next September.  Should I be experiencing speed issues with v1.17 of GiA?

**Comment by skovatch** [comment]
The only changes I've had to make for 10.15 is that iTunes no longer exists, so I need to find a different app (Music.app or TV.app) for adding files. It is compiled in Xcode 11.3 with the latest SDK, but its compatibility version is 10.10.

However, I don't see how any of that matters since it's get_iplayer or youtube-dl doing the downloading, and those are launched into their own tasks running perl or python.

If it's not too much trouble, please try installing get_iplayer or youtube-dl and download a show that way. See if there's a difference in download speeds on the command line compared to GiA.

**Comment by sheepworrier** [comment]
I used the youtube-dl application within GiA and it worked without any errors on the same ITV episode I’d tried (and failed) from GiA on two different Macs.

I ran in verbose mode in GiA and copied the args to the command line for youtube-dl and it worked. Weird, eh?

**Comment by espaguete-code** [comment]
i'm using smart dns outside uk and britain's got talkent has taken 45 minutes to download 33% - that feels like an awfully long time to me. I am using macbook with 10.15.6.

Anyone similar but getting faster downloads?
